CMakeLists.txt

```diff
@@ -1,90 +1,118 @@
-    target_link_libraries(${PROJECT_NAME}-test_exact_time_policy message_filters ${GTEST_LIBRARIES})
+  ament_add_gtest(${PROJECT_NAME}-test_chain test/test_chain.cpp)
+  if(TARGET ${PROJECT_NAME}-test_chain)
+    target_link_libraries(${PROJECT_NAME}-test_chain ${PROJECT_NAME})
   endif()
 
-  catkin_add_gtest(${PROJECT_NAME}-test_approximate_time_policy test/test_approximate_time_policy.cpp)
-  if(TARGET ${PROJECT_NAME}-test_approximate_time_policy)
-    target_link_libraries(${PROJECT_NAME}-test_approximate_time_policy message_filters ${GTEST_LIBRARIES})
+  ament_add_gtest(${PROJECT_NAME}-test_subscriber test/test_subscriber.cpp)
+  if(TARGET ${PROJECT_NAME}-test_subscriber)
+    target_link_libraries(${PROJECT_NAME}-test_subscriber ${PROJECT_NAME})
   endif()
 
-  catkin_add_gtest(${PROJECT_NAME}-test_simple test/test_simple.cpp)
-  if(TARGET ${PROJECT_NAME}-test_simple)
-    target_link_libraries(${PROJECT_NAME}-test_simple message_filters ${GTEST_LIBRARIES})
+  ament_add_gtest(${PROJECT_NAME}-test_synchronizer test/test_synchronizer.cpp)
+  if(TARGET ${PROJECT_NAME}-test_synchronizer)
+    target_link_libraries(${PROJECT_NAME}-test_synchronizer ${PROJECT_NAME})
   endif()
 
-  catkin_add_gtest(${PROJECT_NAME}-test_chain test/test_chain.cpp)
-  if(TARGET ${PROJECT_NAME}-test_chain)
-    target_link_libraries(${PROJECT_NAME}-test_chain message_filters ${GTEST_LIBRARIES})
+  ament_add_gtest(${PROJECT_NAME}-time_synchronizer_unittest test/time_synchronizer_unittest.cpp)
+  if(TARGET ${PROJECT_NAME}-time_synchronizer_unittest)
+    target_link_libraries(${PROJECT_NAME}-time_synchronizer_unittest ${PROJECT_NAME})
   endif()
 
-  # Needs to be a rostest because it spins up a node, which blocks until it hears from the master (unfortunately)
-  add_rostest_gtest(${PROJECT_NAME}-time_sequencer_unittest test/time_sequencer_unittest.xml test/time_sequencer_unittest.cpp)
-  if(TARGET ${PROJECT_NAME}-time_sequencer_unittest)
-    target_link_libraries(${PROJECT_NAME}-time_sequencer_unittest message_filters)
+  ament_add_gtest(${PROJECT_NAME}-test_exact_time_policy test/test_exact_time_policy.cpp)
+  if(TARGET ${PROJECT_NAME}-test_exact_time_policy)
+    target_link_libraries(${PROJECT_NAME}-test_exact_time_policy ${PROJECT_NAME})
   endif()
 
-  add_rostest_gtest(${PROJECT_NAME}-test_subscriber test/test_subscriber.xml test/test_subscriber.cpp)
-  if(TARGET ${PROJECT_NAME}-test_subscriber)
-    target_link_libraries(${PROJECT_NAME}-test_subscriber message_filters)
+  ament_add_gtest(${PROJECT_NAME}-test_approximate_time_policy test/test_approximate_time_policy.cpp)
+  if(TARGET ${PROJECT_NAME}-test_approximate_time_policy)
+    target_link_libraries(${PROJECT_NAME}-test_approximate_time_policy ${PROJECT_NAME})
   endif()
 
-  # Unit test of the approximate synchronizer
-  catkin_add_nosetests(test/test_approxsync.py)
-  catkin_add_nosetests(test/test_message_filters_cache.py)
+#  ament_add_gtest(${PROJECT_NAME}-test_fuzz test/test_fuzz.cpp)
```

Comment:
If the fuzz test defaults to disable, it's better deal with the  ROS2 way by `SKIP_TEST` with a macro to bypass it, then enabling it by give the macro defined to `--cmake-args` when necessary, and I can do this work later.